### ISSUE TITLE
Check if the city exists before adding

### ISSUE THREAD

**Issue by Aatmaj-Zephyr** [Aatmaj-Zephyr/Salseman-2.0#7]
Prevent duplication.

**Comment by sgaurav37533** [comment]
I can work on this please assign me this issue.

**Comment by priyanka010392** [comment]
Hello @Aatmaj-Zephyr , could you please assign it to me? I would like to work on it during Hacktober fest

**Comment by Aatmaj-Zephyr** [comment]
I have assigned you both. Thanks for the help. Although I do not understand how this issue will be fixed before the frontend and backend are coupled.

**Comment by Aatmaj-Zephyr** [comment]
This can be solved at the backend. When the duplicate city is placed, it is not added to the database. (Currently the file)
This can be done by making a list of cities and checking each one. We will need to do that anyway for the other parts of the code. 

Another alternative that can be done is that if the city duplicates are present, then the error will be given to the frontend. But this not only will require additional code and network resources but also not appropriate in terms of UI. The user should not feel 'bad' unnecessarily. The user doesn't need to know that he has done double work and added a city already added before.

**Comment by Aatmaj-Zephyr** [comment]
I have fixed it from the front end side. The `city` list already loads the values of the cities.  So before submitting the form, this is checked and if the city is already present, then it won't be added. However, the best thing will be to reduce the load at front end and let the backend handle it.  I have added error anyways as I have taken this alternative.

So you can fix it on that end and make code better by the operations in backend

**Comment by priyanka010392** [comment]
> I have fixed it from the front end side. The `city` list already loads the values of the cities. So before submitting the form, this is checked and if the city is already present, then it won't be added. However, the best thing will be to reduce the load at front end and let the backend handle it. I have added error anyways as I have taken this alternative.
> 
> So you can fix it on that end and make code better by the operations in backend

Right, it would be better to have this check in the backend instead of frontend.

**Comment by priyanka010392** [comment]
@Aatmaj-Zephyr The backend code is full of compilation errors, doesn't even run properly.

**Comment by Aatmaj-Zephyr** [comment]
@priyanka010392 It compiles correctly from the main branch. From the other branches, some stubs are present

**Comment by Aatmaj-Zephyr** [comment]
I suppose there is one corrupted file. I have fixed that in the main branch. If still error persists, you can find the entire backend code at the repository [here](https://github.com/Aatmaj-Zephyr/Travelling-salesman/tree/main/src)

**Comment by priyanka010392** [comment]
> I suppose there is one corrupted file. I have fixed that in the main branch. If still error persists, you can find the entire backend code at the repository [here](https://github.com/Aatmaj-Zephyr/Travelling-salesman/tree/main/src)

The other repo works, thanks!

**Comment by Aatmaj-Zephyr** [comment]
How did you run it? I copy-paste each file from the online editor. That was the root cause of some files with mismatched contents.

**Comment by priyanka010392** [comment]
let me fix the backend code in this repo, need to replace the contents of this repo(only backend folder) with the files from the travelling salesman repo.

**Comment by priyanka010392** [comment]
Shall I create a separate issue for the above task and then we can start with the duplicate cities issue?

**Comment by Aatmaj-Zephyr** [comment]
Sure. You can do that in the main branch. Yes, you can create a new issue thanks.
